packages/compass-generative-ai/src/atlas-ai-service.ts

```diff
@@ -437,6 +442,30 @@ export class AtlasAiService {
     );
   }
 
+  // optIn function w post request to set the user preference to true
+  async optIntoGenAIFeaturesAtlas() {
+    const res = await this.atlasService.authenticatedFetch(
+      this.atlasService.cloudEndpoint(
+        '/settings/optInDataExplorerGenAIFeatures'
+      ),
+      {
+        method: 'POST',
+        body: JSON.stringify({
+          isEnabled: true,
+        }),
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+        },
+      }
+    );
+    if (res.ok) {
+      await this.preferences.savePreferences({
+        optInDataExplorerGenAIFeatures: true,
+      });
+    }
```

Comment:
Should we be handling when the `res.ok` isn't true? Looks like right now it would still be performing `OptInSuccess`